ros2param/test/test_verb_dump.py

```diff
@@ -109,52 +146,83 @@ def setUp(self):
+        assert launch_testing.tools.expect_output(
+            expected_lines=["'invalid_path' is not a valid directory."],
+            text=param_dump_command.output,
+            strict=True
+        )
 
     def test_verb_dump(self):
         with tempfile.TemporaryDirectory() as tmpdir:
-            assert cli.main(
-                    argv=['param', 'dump', '/foo/test_node', '--output-dir', tmpdir]) is None
-
+            with self.launch_param_dump_command(
+                arguments=[f'{TEST_NAMESPACE}/{TEST_NODE}', '--output-dir', tmpdir]
+            ) as param_dump_command:
+                assert param_dump_command.wait_for_shutdown(timeout=TEST_TIMEOUT)
+            assert param_dump_command.exit_code == launch_testing.asserts.EXIT_OK
+            assert launch_testing.tools.expect_output(
+                expected_lines=[''],
+                text=param_dump_command.output,
+                strict=True
+            )
             # Compare generated parameter file against expected
             generated_param_file = os.path.join(tmpdir, self._output_file())
             assert (open(generated_param_file, 'r').read() == EXPECTED_PARAMETER_FILE)
 
     def test_verb_dump_print(self):
-        with patch('sys.stdout', new=StringIO()) as fake_stdout:
-            assert cli.main(
-                argv=['param', 'dump', 'foo/test_node', '--print']) is None
-
-            # Compare generated stdout against expected
-            assert fake_stdout.getvalue().strip() == EXPECTED_PARAMETER_FILE[:-1]
-
+        with self.launch_param_dump_command(
+            arguments=[f'{TEST_NAMESPACE}/{TEST_NODE}', '--print']
+        ) as param_dump_command:
+            assert param_dump_command.wait_for_shutdown(timeout=TEST_TIMEOUT)
+        assert param_dump_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_text=EXPECTED_PARAMETER_FILE + '\n',
+            text=param_dump_command.output,
+            strict=True
+        )
+        # If both '--output-dir' and '--print' options are provided, ensure it only prints
+        # and no file is written
         with tempfile.TemporaryDirectory() as tmpdir:
-            assert cli.main(
-                argv=['param', 'dump', 'foo/test_node', '--output-dir', tmpdir, '--print']) is None
+            with self.launch_param_dump_command(
+                arguments=[f'{TEST_NAMESPACE}/{TEST_NODE}', '--output-dir', tmpdir, '--print']
+            ) as param_dump_command:
+                assert param_dump_command.wait_for_shutdown(timeout=TEST_TIMEOUT)
+            assert param_dump_command.exit_code == launch_testing.asserts.EXIT_OK
+            assert launch_testing.tools.expect_output(
+                expected_text=EXPECTED_PARAMETER_FILE + '\n',
+                text=param_dump_command.output,
+                strict=True
+            )
 
             not_generated_param_file = os.path.join(tmpdir, self._output_file())
-
```

Comment:
Are these whitespace changes are intentional cleanup or artifacts?

Reply:
Intentional. I thought it looked better this way; grouping together the LOC related to checking that the directory does not exist. It's subjective, and I can revert if you prefer.